### PR TITLE
Add visitor feature tests

### PR DESCRIPTION
**Pull request by lucyfox4131**
add visitor test so that a visitor can add a sock
to their cart and navigate to that cart to see what
they should see.

### CODE REVIEW

**Comment by lucyfox4131** [comment]
Have fixed the hound complaints regarding single quotes. This will be seen on the next pull request.

File: spec/features/visitor_can_create_account_and_get_to_their_dashboard_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      fill_in :name, with: "Bob"
+      fill_in :username, with: "username"
+      fill_in :password, with: "password"
+      click_on "Create New Account"
+
+      expect(current_path).to eq("/dashbaord")
+
+      within "nav" do
+        expect(page).to have_content("Logged in as Bob")
+        #expect page to have profile information!
```
Comment (marcellawigg) [inline spec/features/visitor_can_create_account_and_get_to_their_dashboard_spec.rb]: dashboard spelled wrong; test will fail

File: spec/features/visitor_removes_item_from_their_cart_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    end
+  end
+end
+
+# Background: My cart has an item in it
+
+# Then my current page should be "/cart"
+# And I should see a message styled in green
+# And the message should say "Successfully removed SOME_ITEM from your cart."
+# And the title "SOME_ITEM" should be a link to that item in case the user wants to add it back
```
Comment (ckaminer) [inline spec/features/visitor_removes_item_from_their_cart_spec.rb]: "Line too long" from hound is referring to commented out user story.  Remove before pushing to master.

File: spec/features/visitor_can_create_account_and_get_to_their_dashboard_spec.rb (hunk 1 above)
Comment (ckaminer) [inline spec/features/visitor_can_create_account_and_get_to_their_dashboard_spec.rb]: Hound comment referring to user story

File: spec/features/visitor_can_adjust_quantity_of_an_item_in_cart_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    item = create(:item)
+    add_to_cart(item, 3)
+
+    visit "/cart"
+
+    within '#cart-sock' do
+      expect(page).to have_content(sock.name)
+      expect(page).to have_content("Quantity: 3")
+    end
+    # may need to be "option[1]" instead of just 1, right now this is written for a drop down menu
```
Comment (ckaminer) [inline spec/features/visitor_can_adjust_quantity_of_an_item_in_cart_spec.rb]: Need to remove commented out user stories before pushing to master.

**Pull request merged**